components/calendar/calendar.js

```diff
@@ -551,6 +581,23 @@ class Calendar extends LocalizeCoreElement(RtlMixin(LitElement)) {
 		});
 	}
 
+	willUpdate(changedProperties) {
+		super.willUpdate(changedProperties);
+		if (changedProperties.get('_shownYear') === undefined && changedProperties.get('_shownMonth') === undefined) return;
+
+		const dates = getDatesInMonthArray(this._shownMonth, this._shownYear);
```

Comment:
There is a bit of work that `getDatesInMonthArray(...)` does... nice that it is usable. 😄  An alternative is to just provide the consumer with `_shownYear` and `_shownMonth`, but giving them specific dates means they can be more selective in the events they add to the HTML.